client/src/bastionlab/polars/policy.py

```diff
@@ -1,14 +1,14 @@
 from dataclasses import dataclass
-from typing import List
+from typing import List, Union
+from serde import serde, InternalTagging
 
 
 class Rule:
-    def serialize(self) -> str:
-        raise NotImplementedError
+    pass
```

Comment:
I don't think we need to subtype an abstract class anymore.
Instead, a uniform approach to use Unions everywhere would be to use a typevar equal to the union. This is event closer and truer to what enums are!